mypy/argmap.py

```diff
@@ -162,16 +167,21 @@ def expand_actual_type(self,
         This is supposed to be called for each formal, in order. Call multiple times per
         formal if multiple actuals map to a formal.
         """
+        from mypy.subtypes import is_subtype
+
         actual_type = get_proper_type(actual_type)
         if actual_kind == nodes.ARG_STAR:
-            if isinstance(actual_type, Instance):
-                if actual_type.type.fullname == 'builtins.list':
-                    # List *arg.
-                    return actual_type.args[0]
-                elif actual_type.args:
-                    # TODO: Try to map type arguments to Iterable
+            if isinstance(actual_type, Instance) and actual_type.args:
+                if is_subtype(actual_type, self.context.iterable_type):
+                    # TODO: this can be invalid, when generic type has:
+                    # `class Some(Generic[X, Y], Iterable[Y]):`
+                    # https://github.com/python/mypy/issues/11138
```

Comment:
Can you use `map_instance_to_supertype` to work around the issue?

Reply:
I've planned to properly work on it after this initial PR.
I've found several corner-cases that are out of scope of this piece of work.

But, if that's a problem I can just add it all at once 🙂

Reply:
Done! 👍